### PR TITLE
ARROW-9674: [Rust] Make the parquet read and writers Send

### PR DESCRIPTION
**Pull request by Marwes**
Based on #8509

### CODE REVIEW

**Comment by nevi-me** [comment]
@Marwes may you please rebase this, I think the conflicts are from your other PR, which might share some commits with this one

**Comment by nevi-me** [comment]
The memory tracker changes are failing. Was there a big that you picked up @Marwes? Else we could isolate that, and leave the `Rc` > `Arc` changes, which I'm happy with

**Comment by Marwes** [comment]
The `Cell` needs to be removed to make the buffer types `Send`. Anyway, it is fixed.

**Comment by alamb** [comment]
I will try and review this either later today or tomorrow morning.

**Review by nevi-me — APPROVED** [review]
LGTM

**Review by sunchao — APPROVED** [review]
Makes sense. LGTM.

File: rust/parquet/src/util/memory.rs [hunk 1]
```diff
@@ -39,37 +41,39 @@ pub type WeakMemTrackerPtr = Weak<MemTracker>;
...
     #[inline]
     pub fn alloc(&self, num_bytes: i64) {
-        let (current, mut maximum) = self.memory_usage.get();
-        let new_current = current + num_bytes;
-        if new_current > maximum {
-            maximum = new_current
-        }
-        self.memory_usage.set((new_current, maximum));
+        let new_current = self
+            .current_memory_usage
```
Comment (alamb) [inline rust/parquet/src/util/memory.rs:72]: 👍  This is a nice change to remove the use of `Cell`

**Review by alamb — APPROVED** [review]
Looks great. Nice work @Marwes

**Comment by nevi-me** [comment]
Beat me to the merge @alamb by seconds

**Comment by alamb** [comment]
> Beat me to the merge @alamb by seconds


🏃 Sorry @nevi-me !

**Pull request closed without merging**